src/connectors/ipfs/index.ts

```diff
@@ -1,17 +1,53 @@
+import { generateKeyPair } from 'crypto'
+import FormData from 'form-data'
 import { create } from 'ipfs-http-client'
+import fetch from 'node-fetch'
+import { Readable } from 'stream'
+import { promisify } from 'util'
+
+const generateKeyPairPromisified = promisify(generateKeyPair)
 
 import { environment } from 'common/environment'
 
+const ipfsServerUrl = `http://${environment.ipfsHost}:${environment.ipfsPort}/api/v0`
+
 export class IPFS {
   client: any
 
   constructor() {
     this.client = create({
-      host: environment.ipfsHost,
-      port: parseInt(environment.ipfsPort, 10),
-      protocol: 'http',
+      // host: environment.ipfsHost,
+      // port: parseInt(environment.ipfsPort, 10),
+      // protocol: 'http',
+      url: ipfsServerUrl,
     })
   }
+
+  genKey = async () => generateKeyPairPromisified('ed25519') // {
+  // this.client.key.gen(name)
+  // const key = await generateKeyPairPromisified('ed25519')
+  // return key.privateKey.export({ format: 'pem', type: 'pkcs8', // cipher: 'aes-256-cbc', passphrase:'pass1 top secret', })
+  // }
+
+  importKey = async (name: string, pem: string) => {
+    const formData = new FormData()
+    const url = new URL(`${ipfsServerUrl}/key/import`)
+    url.searchParams.set('arg', name)
+    url.searchParams.set('format', 'pem-pkcs8-cleartext')
+    formData.append('file', Readable.from([pem]), 'keyfile')
+    const res = await fetch(url, {
+      method: 'POST',
+      body: formData,
+    })
+    console.log(new Date(), 'key/import res:', res)
```

Comment:
Is there any sensitive data that needs to be masked?

Reply:
no; the sample looks like this:

```js
2022-09-12T18:42:59.223Z key/import fetch res: {
  Name: 'ed-gen-test23-4-1663008179217',
  Id: 'k51qzi5uqu5dhdb6i7umgq6xcu5wt4appzz1c1kyqgh8folsecogs56j11if8n'
}
```

Reply:
many console log lines for debugging in the beginning days when running on dev; but eventually (in 2 ~ 4 weeks range) will be cleaned up